### PR TITLE
Update OtaPalWriteBlock_t comment with a note about returning from the function and exception condition.

### PR DESCRIPTION
**Pull request by ActoryOu**
Update OtaPalWriteBlock_t comment with a note about returning from the function and exception condition.

Description
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by pvyawaha — DISMISSED** [review]

File: source/ota.c [hunk 1]
```diff
@@ -2567,6 +2572,39 @@ static IngestResult_t processDataBlock( OtaFileContext_t * pFileContext,
     return eIngestResult;
 }
 
+/* Store the data block in file context. */
+static IngestResult_t writeDataBlockToFile( OtaFileContext_t * pFileContext,
```
Comment (paulbartell) [inline source/ota.c:2576]: Is a separate function really necessary here?

Comment (ActoryOu) [reply source/ota.c:2576]: If I add an assertion in original processDataBlock function, the complexity level is 9.
So I add this separate function to reduce the level.

Comment (ActoryOu) [reply source/ota.c:2576]: If I add an assertion in original processDataBlock function, the complexity level is 9.
So I add this separate function to reduce the level.

Comment (paulbartell) [reply source/ota.c]: I pushed some changes to address this. Please look them over.

Comment (ActoryOu) [reply source/ota.c]: Thanks. That's great!
I also added an unit test case for the coverage.

**Review by paulbartell — APPROVED** [review]

**Review by pvyawaha — APPROVED** [review]

**Pull request merged**